### PR TITLE
[Server] - API interface - Implementation

### PR DESCRIPTION
**Pull request by thisab**
**Pull Request check list**

- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

**Description of change**
Defined the Server and made the Server implement (dummy initially) functions for the API

**Which issue this PR fixes**

### CODE REVIEW

File: pkg/server/api/management/handler.go [hunk 1]
```diff
@@ -1,35 +1,132 @@
 package management
 
 import (
-	"errors"
-
 	"github.com/labstack/echo/v4"
 )
 
-type MyDumbServer struct {
+type Server struct {
```
Comment (mchurichi) [inline pkg/server/api/management/handler.go:7]: can we define the actual interface this struct implements?

File: pkg/server/api/management/handler.go [hunk 2]
```diff
@@ -1,35 +1,132 @@
...
+	return nil
+}
+
+// (GET /organizations/{orgID})
+func (server Server) GetOrgbyID(ctx echo.Context, orgID int64) error {
+	return nil
+}
+
+// (PUT /organizations/{orgID})
+func (server Server) UpdateOrganizaion(ctx echo.Context, orgID int64) error {
```
Comment (n-th) [inline pkg/server/api/management/handler.go:130]: UpdateOrganization

File: pkg/server/api/management/handler.go [hunk 3]
```diff
@@ -1,35 +1,132 @@
...
+	return nil
+}
+
+// (GET /federationRelationships/{relationshipID})
+func (server Server) GetFederationRelationshipbyID(ctx echo.Context, relationshipID int64) error {
+	return nil
+}
+
+// (PUT /federationRelationships/{relationshipID})
+func (server Server) UpdateFederationRelationshipship(ctx echo.Context, relationshipID int64) error {
```
Comment (n-th) [inline pkg/server/api/management/handler.go:105]: UpdateFederationRelationship

File: pkg/server/api/management/handler.go [hunk 4]
```diff
@@ -1,35 +1,132 @@
...
+	return nil
+}
+
+// (POST /federationGroupMemberships)
+func (server Server) CreateFederationGroupMembership(ctx echo.Context) error {
+	return nil
+}
+
+// (DELETE /federationGroupMemberships/{membershipID})
+func (server Server) DeletefederationGroupMembership(ctx echo.Context, membershipID int64) error {
```
Comment (n-th) [inline pkg/server/api/management/handler.go:50]: DeleteFederationGroupMembership

Comment (n-th) [reply pkg/server/api/management/handler.go:50]: This seems to happen to all federationGroupMembership related functions

**Review by n-th — CHANGES_REQUESTED** [review]
I found some typos in the function, could you revise the .yamls that generate those functions for misspells?

File: pkg/server/api/management/handler.go (hunk 2 above)
Comment (thisab) [reply pkg/server/api/management/handler.go:130]: Indeed. To do that, we would need to include management.yaml file and correct it also.

File: pkg/server/api/management/handler.go [hunk 5]
```diff
@@ -1,35 +1,132 @@
...
+}
+
+func NewServer() *Server {
+	return &Server{}
 }
 
 // (GET /spireServers)
-func (server MyDumbServer) GetSpireServers(ctx echo.Context, params GetSpireServersParams) error {
-	return errors.New("Ah, não, deu erro")
+func (server Server) GetSpireServers(ctx echo.Context, params GetSpireServersParams) error {
```
Comment (maxlambrecht) [inline pkg/server/api/management/handler.go:15]: What's the reason for not using the `context.Context` from the standard library?

Comment (thisab) [reply pkg/server/api/management/handler.go:15]: @maxlambrecht we are using context of the Echo library, following just like the main.go does nowadays.
Do you think we should not follow that ?

Comment (jufantozzi) [reply pkg/server/api/management/handler.go:15]: AFAIK this is how the lib works, this `ctx` will have the request details

Comment (maxlambrecht) [reply pkg/server/api/management/handler.go:15]: We should use the `context.Context` from the standard library in all our APIs definitions. Our APIs should be as standard as possible, if in our implementation we use some library that needs a specific context struct, we'll need to "adapt" the standard ctx to it, but that specific context is an implementation detail that must not be leaked into the public interface.

Comment (wiliamsouza) [reply pkg/server/api/management/handler.go]: I was doing a little research to discover a way to keep using the echo as our web framework but it seems like there is no way to use it without using it’s  own context (`echo.Context` implementation[1]). Searching around I have found out that what other people do is to adapt/convert from `echo.Context` to Go lang `context.Context` before calling downstream code. This way it decouples the inner layers from your web framework API code. The main benefits of this approach is to isolate/decouple web frameworks or ORM from Galadriel  main business logic. May I upload the suggested change to this PR showing that solution?

PS: Maybe use the Go lang `internal` package to enforce that isolation[3] can be an option.

[1] https://stackoverflow.com/questions/47843914/transforming-echo-context-to-context-context
[2] https://github.com/labstack/echo/issues/1175#issuecomment-455226401
[3] https://dave.cheney.net/2019/10/06/use-internal-packages-to-reduce-your-public-api-surface

Comment (maxlambrecht) [reply pkg/server/api/management/handler.go]: You are right, it seems that we need to use the `echo.Context`.

**Comment by thisab** [comment]
@jufantozzi I think this PR is a bit outdated. Does it still make sense to leave it open?

**Comment by jufantozzi** [comment]
@thisab you're right, we can close this

**Pull request closed without merging**